src/components/pages/Dashboard/Dashboard.tsx

```diff
@@ -1,13 +1,32 @@
-import React from "react";
+import React, { useContext, useEffect } from "react";
 import { useMediaQuery } from "react-responsive";
 import { mobileDeviceOrTabletWidth } from "../../../constants";
 import MobileComponents from '../../mobile/MobileComponents';
 import LeftSidebar from "../../sidebar/left-sidebar/LeftSidebar";
 import RightSidebar from "../../sidebar/right-sidebar/RightSidebar";
 import CentralPiece from "./CenterComponent";
+import { AppContext } from "../../../context";
+import httpClient from "../../../httpClient";
+import { AggregationFactor } from "../../../types";
 
 export default function Dashboard() {
-  const isMobileDeviceOrTablet = useMediaQuery({ maxDeviceWidth: mobileDeviceOrTabletWidth })
+  const isMobileDeviceOrTablet = useMediaQuery({ maxDeviceWidth: mobileDeviceOrTabletWidth });
+  const [state, dispatch] = useContext(AppContext);
+  
+  useEffect(() => {
+    if(state.filtered_records.length > 0){
+      const updateCountryPrevalence = async () => {
+        const api = new httpClient();
+        const countryPrevalences = await api.postMetaAnalysis(state.filtered_records, AggregationFactor.country);
+        dispatch({
+          type: 'UPDATE_COUNTRY_PREVALENCES',
+          payload: countryPrevalences
+        });
+      } 
+      updateCountryPrevalence();

```

Comment:
isn't there a chance that this doesn't exist?

Reply:
chance that what doesn't exist

Reply:
nvm